src/net.cpp

```diff
@@ -2556,6 +2550,25 @@ bool CConnman::DisconnectNode(const std::string& strNode)
     }
     return false;
 }
+
+bool CConnman::DisconnectNode(const CSubNet& subnet)
+{
+    bool disconnected = false;
+    LOCK(cs_vNodes);
+    for (CNode* pnode : vNodes) {
+        if (subnet.Match((CNetAddr)pnode->addr)) {
+            pnode->fDisconnect = true;
+            disconnected = true;
```

Comment:
if (disconnected == false) {
    disconnected = true;
}      

To add a little bit of performance